src/librustc_mir/transform/const_prop.rs

```diff
@@ -525,27 +542,39 @@ impl<'a, 'mir, 'tcx> ConstPropagator<'a, 'mir, 'tcx> {
             true,
         ).expect("value should already be a valid const");
 
-        if let interpret::Operand::Immediate(im) = *value {
-            match im {
+        match *value {
+            interpret::Operand::Immediate(im) => match im {
                 interpret::Immediate::Scalar(ScalarMaybeUndef::Scalar(scalar)) => {
-                    *rval = Rvalue::Use(self.operand_from_scalar(scalar, value.layout.ty, span));
+                    *rval = Rvalue::Use(
+                        self.operand_from_scalar(scalar, value.layout.ty, source_info.span));
                 },
                 Immediate::ScalarPair(
                     ScalarMaybeUndef::Scalar(one),
                     ScalarMaybeUndef::Scalar(two)
-                ) => {
+                    ) => {
```

Comment:
The indentation is a bit weird.

Reply:
It seems is vim playing with the indentation, I'll try to fix it

Reply:
It reappeared after your new commits.